Dockerfile

```diff
@@ -56,7 +56,10 @@ RUN mkdir /opt/litecoin && cd /opt/litecoin \
 ENV LIGHTNINGD_VERSION=master
 
 WORKDIR /opt/lightningd
-COPY . .
+COPY . /tmp/lightning
+RUN git clone --recursive /tmp/lightning . && \
+    git checkout $(git --work-tree=/tmp/lightning --git-dir=/tmp/lightning/.git rev-parse HEAD) && \
+    rm -rf /tmp/lightning
```

Comment:
`rm -rf /tmp/lightning/.git && rm -rf /tmp/lightning` works. `rm -rf` does not delete hidden folders maybe? Strange.

Reply:
Shouldn't change the outcome though, since this is just in the builder image which gets discarded anyway.

Reply:
It does, because `rm -rf /tmp/lightning` is returning an error. (see https://circleci.com/gh/NicolasDorier/lightning/20 which fail the build)

Reply:
Could it be some sort of concurrent access? iirc `rm -rf` still recursively deletes things, so if it might be that something new was created inside those directories it can fail.